core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -279,6 +289,20 @@ public static Future<?> addItemToPlaybackHistory(final FeedMedia media) {
         });
     }
 
+    public static Future<?> addItemToPlaybackHistory(final FeedMedia media, Date date) {
```

Comment:
How about extending the existing method (to reduce code duplication), such that the date can be passed as a parameter but doesn't has to? Eg adding a second method that simply calls the first one and adds the parameter.